ap_src/ap_app/utils/switch_permissions.py

```diff
@@ -4,6 +4,8 @@
 from .teams_utils import get_users_sharing_teams
 from .user_profile import get_user_id_from_username, get_userprofile_id_from_user_id
 
+NO_ERRORS = True
+
 def check_for_lingering_switch_perms(username, action):
```

Comment:
I think we need to raise a separate issue to tidy up the utils files. There's a few things I'll note: 

1. The function names don't always describe the functionality. If I call a function called check_for_lingering_switch_perms, I would expect it to return true if there are lingering perms, and false if there aren't. Whereas this returns true/false based on whether any errors occur or not.
2. Relating to the above point, some better error handling would be nice just to be able to properly output what caused the error. Was it because we couldn't find the user id? or was it because of another problem? This helps to quickly debug the problem if and when they do occur if we have specific error messages.
3. Passing an action to the function makes the function more difficult to understand what is going again. This relates to the first point in that the function is doing more than I expect it to. But the way I would write this is that wherever you're calling check_for_lingering_switch_perms from, I would expect to get a result back on whether or not there were any lingering perms, and if there were any, I would then call another function to delete those lingering perms. It makes for more readable code that is easy to understand for someone else looking at it.

Reply:
Points 1. and 3.
I agree. A potential solution I thought of during development was to get `check_for_lingering_switch_perms` to return a data structure of some sort that has the contents of the permissions whitelist table, which is then processed by another function, but this seemed more complex at the time. This would also allow one of the other features to be implemented (i.e., notify the user what permissions are being removed, or even potentially sort through and keep some of them).

I think that since the solution already works and this issue requires some moderate code refactoring, it should be made a separate issue.

Reply:
Point 2.
I'm a little confused by what you mean. The solution does print a specific message to the terminal when an error is encountered.

Reply:
Jayden has pointed out that the error handling is done in the functions that are called by check_for_lingering_switch_perms.